compiler/IREmitter/sends.cc

```diff
@@ -457,6 +457,18 @@ llvm::Value *IREmitterHelpers::emitMethodCallViaRubyVM(MethodCallContext &mcctx)
     auto &irctx = mcctx.irctx;
     auto rubyBlockId = mcctx.rubyBlockId;
 
+    // If we get here with <Magic>, then something has gone wrong.
+    // TODO(froydnj): We want to do the same thing with Sorbet::Private::Static,
+    // but we'd need to do some surgery on either a) making those methods not
+    // be emitted via symbol-based intrinsics or b) making it possible for
+    // (some) symbol-based intrinsics to bypass typechecks completely.
+    if (auto *at = core::cast_type<core::AppliedType>(send->recv.type)) {
+        if (at->klass == core::Symbols::MagicSingleton()) {
+            failCompilation(cs, core::Loc(irctx.cfg.file, send->receiverLoc),
+                            "No runtime implemention for <Magic> method exists");
```

Comment:
Can we interpolate the magic method's name into the error message here?

Reply:
Unfortunately not at the moment, because of how `failCompilation` is defined:

https://github.com/sorbet/sorbet/blob/b4d6348ae968f44c2de13a3dff0c2318589e7956/compiler/Core/CompilerState.h#L71-L74

We can change that in a followup, though?